src/utils/request.ts

```diff
@@ -1,16 +1,37 @@
-import { DocumentNode } from 'graphql';
-import gql from 'graphql-tag';
-import { getKeyForRequest } from './keyForQuery';
+import { DocumentNode, parse, print } from 'graphql';
+import { hash, phash } from './hash';
+import stringify from 'fast-json-stable-stringify';
 import { GraphQLRequest, Operation, OperationContext } from '../types';
 
+interface Documents {
+  [key: number]: DocumentNode;
+}
+
+const docs: Documents = Object.create(null);
+const keyProp = '__key';
+
 export const createRequest = (
   q: string | DocumentNode,
   vars?: object
 ): GraphQLRequest => {
-  const query = typeof q === 'string' ? gql([q]) : q;
+  let key: number;
+  let query: DocumentNode;
+  if (typeof q === 'string') {
+    key = hash(q.replace(/[\s,]+/, ' ').trim());
```

Comment:
If I'm reading this correctly, this RegExp will just replace the first whitespace character or `,` it encounters (or both if they're adjacent) with a whitespace character. Can you explain a bit more what purpose that's serving? I'm reading through the `hash` code above to try to get more context but admittedly bitwise operators are not my strong suit .

Reply:
This is borrowed from graphql-tag itself and meant to stabilise hashes. Differently formatted queries will yield the same hash when the formatting is applied, so the intention is for two queries to have the same hash, even when they've been parsed & stringified or formatted through other means

Reply:
> this RegExp will just replace the **first** whitespace character or `,` it encounters

You're right, I misread your comment. Sorry! That was a copy&paste issue. I've just fixed it in the latest commit